### ISSUE TITLE
ENH: Cleaner Conditional Column Selection 

### ISSUE THREAD

**Issue by madisonestabrook** [pandas-dev/pandas#45569]
#### Is your feature request related to a problem?
I wish I could cleanly and concisely conditionally select columns that I need in a pandas dataframe.

#### Describe the solution you'd like
```python
"""
A function to only select the columns that the user needs, based on simple string methods (`starts_with`, `ends_with`, and `in`)
"""
```
#### API breaking implications
I don't foresee any breaking API changes, just easier and cleaner conditional selecting of columns for users.

#### Describe alternatives you've considered
As far I know, users have to use a list comprehension to conditionally select the desired columns. Another option is to manually create a list of the columns. However, the first option is not the most intuitive and the second option is error-prone. 
```python
df = pd.DataFrame([{'a': 0, 'b': 1, 'aa': 2, 'aaaa': 3, 'bbb': 4, 'aaaa': 5, 'cab': 6}]) # Sample Data
df

# Suppose I just need the columns whose first letter is "a". Then, I would do something like the following:
cols = [c for c in df.columns if c.startswith('a')] # ['a', 'aa', 'aaaa']
df[cols]

# Now, suppose I need all columns that contain an "a". Then, I would do something like the following:
cols = [c for c in df.columns if 'a' in c)] # ['a', 'aa', 'aaaa', 'cab']
df[cols]
```
#### Additional context
I would love to work on this issue.

```python
import pandas as pd
pd.show_versions() #pandas 1.1.0
```

**Comment by jreback** [comment]
there is already df.columns.str,startswith() and df.filter 

pls read the docs

**Comment by madisonestabrook** [comment]
Thank you!

I read the docs for [ `pandas.Series.str.startswith`](https://pandas.pydata.org/docs/reference/api/pandas.Series.str.startswith.html) and [`pandas.DataFrame.filter`](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.filter.html). 

Before I submitted this issue, I had searched the docs, both manually and with the search function. I found the options for Series (like `pandas.Series.str.startswith`), but it didn't occur to me to try `pandas.DataFrame.filter` for DataFrames.
